### PR TITLE
Spinning Cube

### PR DESCRIPTION
**Pull request by shurunxuan**
There are some significant changes in this PR:

* Refactored all code to use Vulkan RAII hpp API;
  * Vulkan hpp handles API dynamic loading, so volk is removed;
  * Removed current RAII encapsulation to Vulkan handles as the RAII hpp API is doing a pretty good job on it;
    * Currently all code are written in plain Vulkan (RAII hpp flavor). I'll open another branch if we ever want to support other APIs;
* Added multiplication to Matrix4x4 and removed all redundant operator overloads;
* Since encapsulation is removed for now, #17 and #18 are fixed.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -44,7 +45,7 @@ if (CMAKE_BUILD_TYPE STREQUAL "Debug")
     list(APPEND COMMON_DEFINITIONS "ENGINE_DEBUG=1")
 endif ()
 
-set (SHADER_SOURCE_FILES "Shaders/sample/triangle.hlsl")
+set (SHADER_SOURCE_FILES "Shaders/sample/cube.hlsl")
```
Comment (MarshallW906) [inline src/CMakeLists.txt:48]: Maybe this (and also compiling & copying commands) should be moved into `SampleApp`? i.e. `sample/CMakeLists.txt`

**Review by MarshallW906 — APPROVED** [review]
Good :)
Love the cube

File: src/CMakeLists.txt (hunk 1 above)
Comment (shurunxuan) [reply src/CMakeLists.txt:48]: There are two considerations. One is that the engine itself will need its built-in shaders in the future and this serves as a starting point. The other one is that the code that is using this shader is not currently considered as a "user-space" code, like for now the functionality of the whole "rendering system" is to draw a rotating cube lol

Comment (MarshallW906) [reply src/CMakeLists.txt:48]: Understood. It totally makes sense if is by fact a default resource

**Review by Top61ly — APPROVED** [review]
LGTM!

**Pull request merged**